src/Microsoft.DotNet.Build.Tasks/PackageFiles/versioning.targets

```diff
@@ -151,6 +154,75 @@
     </ItemGroup>
   </Target>
 
+  <!-- #################################### -->
+  <!-- Generate BuildVersion.props -->
+  <!-- #################################### -->
+  <Target Name="CreateOrUpdateCurrentVersionFile" Condition="'$(SkipVersionGeneration)'!='true'" DependsOnTargets="GenerateCurrentVersion">
+    <Error Condition="'$(OverrideBuildNumberMajor)'==''" Text="Error when generating current version." />
+
+    <ItemGroup>
+      <CurrentVersionLines Include="%3C%3Fxml version=%221.0%22 encoding=%22utf-8%22%3F%3E" />
+      <CurrentVersionLines Include="%3C!-- This is a generated file. To re-generate it, run build.cmd --%3E" />
+      <CurrentVersionLines Include="%3CProject xmlns=%22http://schemas.microsoft.com/developer/msbuild/2003%22%3E" />
+      <CurrentVersionLines Include="%3CPropertyGroup%3E" />
+      <CurrentVersionLines Include="%3CBuildNumberMajor Condition=%22%27%24(BuildNumberMajor)%27==%27%27%22%3E$(OverrideBuildNumberMajor)%3C/BuildNumberMajor%3E" />
```

Comment:
Do we expect that this file will always be imported above versioning.targets? Otherwise BuildNumberMajor might be 0 all the time because we set it in there.